### ISSUE TITLE
LSP: add pendant for cobc's `-fmf-comment` and auto-enable it for `$ SET`

### ISSUE THREAD

**Issue by GitMensch** [OCamlPro/superbol-studio-oss#86]
This will fix the syntax error currently raised in: test/testsuite/microfocus/www.csis.ul.ie/Indexed/DirectReadIdx.CBL

**Comment by nberth** [comment]
This issue is actually related to the (current) lack of project configuration in the testsuite. When the LSP/project is configured with `dialect = "mf"` the file actually parses properly.

Once #69 is merged, adding a `superbol.toml` file in `test/testsuite/microfocus/www.csis.ul.ie/` with the following contents should fix the issue:

```toml
[cobol]
dialect = "mf" # or "mf-strict"
```

(Before #69 is merged, the `[cobol]` section header needs to be omitted.)

This makes me realize the LSP server is not verbose enough as to what its current configuration is. Maybe some notification with a summary of relevant configurations could be shown when a file is opened (or the server is restarted).
I'll make that a new issue.

**Comment by GitMensch** [comment]
That new issue is reasonable.

But I do disagree about the necessary configuration. If we are in auto-format and the first thing we get to is a `$` directive we could not only set the format accordingly but also set the "format related" mf-comment (both in cobc and the lsp).

... or one could argue that this is _really_ about the dialect (which we don't have an auto-recognizing) and not about the format.
In this case we should consider adding this auto-dialect thing (again, both to cobc and the lsp). I can come up with a short spec for this.

**Comment by nberth** [comment]
Ah I forgot about the "auto-enable" part.

If GnuCOBOL indeed sets `mf-comment` automatically, then SuperBOL should do as well.

The auto-dialect could be more intricate to specify and implement properly though, as we're already deep down the lexing/pre-processing phase when we guess the format.

**Comment by GitMensch** [comment]
> If GnuCOBOL indeed sets `mf-comment` automatically, then SuperBOL should do as well.

It doesn't. It could by deducing that during deducing the format, but if people pass a -fformat then the deducing won't work any more, so maybe that isn't that clean either.
